### PR TITLE
Add pre-commit config and lint files

### PR DESCRIPTION
**Pull request by klutchell**
Also adding automated management of pre-commit dependencies [here](https://github.com/balena-io/renovate-config/pull/1269).

### CODE REVIEW

File: .github/actions/test/action.yml [hunk 1]
```diff
@@ -77,7 +77,7 @@ runs:
         ip addr
         ip link list
         ip route
-
```
Comment (jakogut) [inline .github/actions/test/action.yml:80]: Noise?

File: .github/workflows/flowzone.yml [hunk 2]
```diff
@@ -25,7 +25,7 @@ permissions:
   repository-projects: none
   security-events: none
   statuses: none
-
```
Comment (jakogut) [inline .github/workflows/flowzone.yml:28]: Noise?

File: runner/s6-overlay/s6-rc.d/dockerd/type [hunk 3]
```diff
@@ -1 +1 @@
-longrun
\ No newline at end of file
+longrun
```
Comment (jakogut) [inline runner/s6-overlay/s6-rc.d/dockerd/type:1]: Noise?

**Review by jakogut — APPROVED** [review]
Looks good

File: .github/actions/test/action.yml (hunk 1 above)
Comment (klutchell) [reply .github/actions/test/action.yml:80]: Intentionally updated by [this](https://github.com/product-os/self-hosted-runners/pull/420/files#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9R10) pre-commit lint rule.

File: runner/s6-overlay/s6-rc.d/dockerd/type (hunk 3 above)
Comment (klutchell) [reply runner/s6-overlay/s6-rc.d/dockerd/type:1]: Intentionally updated by [this](https://github.com/product-os/self-hosted-runners/pull/420/files#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9R11) pre-commit rule.

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+
+# Ignore auto-generated files
+exclude: ^(.versionbot/|CHANGELOG.md|VERSION)
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
```
Comment (joshbwlng) [inline .pre-commit-config.yaml:8]: I see they have a v5 release as of October: https://github.com/pre-commit/pre-commit-hooks/releases
Can we use that or do we need to stay with v3?

Comment (klutchell) [reply .pre-commit-config.yaml:8]: Renovate will automatically bump the revs when new versions are found, assuming the tests pass.

Flowzone uses pre-commit for testing as well, and v5 [merged today](https://github.com/product-os/flowzone/pull/1306).

I can manually bump all of the revs now to latest if you'd prefer, I was just going to let Renovate handle it after the merge so we don't need to do the testing manually.

**Review by joshbwlng — APPROVED** [review]

**Pull request merged**